### PR TITLE
feat: add PartitionedUpdate support to executor

### PR DESCRIPTION
**Pull request by rajatbhatta**
This PR adds support for PartitionedUpdate to Cloud Client Executor Framework.

### CODE REVIEW

**Comment by rajatbhatta** [comment]
The tests will fail until proto changes for PartitionedUpdateAction is made public.

**Comment by rajatbhatta** [comment]
@gyang-google: Can you please review this PR once?

**Comment by gyang-google** [comment]
LGTM

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/Options.java [hunk 1]
```diff
@@ -39,6 +39,13 @@ public enum RpcPriority {
     RpcPriority(Priority proto) {
       this.proto = Preconditions.checkNotNull(proto);
     }
+
+    public static RpcPriority fromProto(Priority proto) {
+      for (RpcPriority e : RpcPriority.values()) {
+        if (e.proto.equals(proto)) return e;
+      }
+      return null;
```
[the remark this answers is not in the document]
Comment (rajatbhatta) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/Options.java]: Added a `RpcPriority.UNSPECIFIED` to enum to map to `Priority.PRIORITY_UNSPECIFIED`. If a `null` value is passed in, we would return `RpcPriority.UNSPECIFIED`.

File: google-cloud-spanner-executor/src/main/java/com/google/cloud/executor/spanner/CloudClientExecutor.java [hunk 2]
```diff
@@ -1974,6 +1984,33 @@ private Status executeExecutePartition(
...
+  /** Execute a partitioned update which runs different partitions in parallel. */
+  private Status executePartitionedUpdate(
+      PartitionedUpdateAction action, DatabaseClient dbClient, OutcomeSender sender) {
+    try {
+      ExecutePartitionedUpdateOptions options = action.getOptions();
+      Long count =
+          dbClient.executePartitionedUpdate(
+              Statement.of(action.getUpdate().getSql()),
+              Options.tag(options.getTag()),
+              Options.priority(RpcPriority.fromProto(options.getRpcPriority())));
```
[the remark this answers is not in the document]
Comment (rajatbhatta) [reply google-cloud-spanner-executor/src/main/java/com/google/cloud/executor/spanner/CloudClientExecutor.java:1996]: RpcPriority.fromProto(null) will return `RpcPriority.UNSPECIFIED`.

**Review by arpan14 — APPROVED** [review]

**Pull request merged**